### PR TITLE
[TECHNICAL SUPPORT] LPS-102947 Fix integration test

### PR DESCRIPTION
**Pull request by rodrigopaulino**
Relevant ticket:
https://issues.liferay.com/browse/LPS-102947

The first PR was not good because it wasn't considering the ResourcePermissions of DDMStructures that don't use a composite name like when you add a DDMFormInstance.

### CODE REVIEW

**Comment by rodrigopaulino** [comment]
ci:test:sf

**Comment by rodrigopaulino** [comment]
ci:test:relevant

**Comment by leoadb** [comment]
Just started reviewing :)

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/search/util/DDMSearchHelper.java [hunk 1]
```diff
@@ -157,6 +172,19 @@ public SearchContext buildTemplateSearchContext(
 		searchContext.setAttribute("language", language);
 		searchContext.setAttribute("mode", mode);
 		searchContext.setAttribute("resourceClassNameId", resourceClassNameId);
```
Comment (leoadb) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/search/util/DDMSearchHelper.java]: @rodrigopaulino should we remove this from the search context? we are not indexing this attribute.

Comment (rodrigopaulino) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/search/util/DDMSearchHelper.java]: Yes, we do index it for DDMTemplate types.

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/search/spi/model/index/contributor/DDMStructureModelDocumentContributor.java [hunk 2]
```diff
@@ -61,15 +62,18 @@ public void contribute(Document document, DDMStructure ddmStructure) {
 
 			document.addKeyword(Field.STATUS, structureVersion.getStatus());
 			document.addKeyword(Field.VERSION, structureVersion.getVersion());
+
+			document.addKeyword(
```
Comment (leoadb) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/search/spi/model/index/contributor/DDMStructureModelDocumentContributor.java]: @rodrigopaulino is the order important? I mean, can we move to before line 63?

Comment (leoadb) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/search/spi/model/index/contributor/DDMStructureModelDocumentContributor.java]: I've moved to other try/catch block, as an exception might be thrown in the previous lines.

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/search/spi/model/index/contributor/DDMStructureModelDocumentContributor.java [hunk 3]
```diff
@@ -61,15 +62,18 @@ public void contribute(Document document, DDMStructure ddmStructure) {
...
+				_ddmPermissionSupport.getStructureModelResourceName(
+					ddmStructure.getClassNameId()));
 		}
 		catch (PortalException pe) {
 			if (_log.isDebugEnabled()) {
 				_log.debug(pe, pe);
 			}
 		}
 
-		document.addKeyword(
```
Comment (leoadb) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/search/spi/model/index/contributor/DDMStructureModelDocumentContributor.java]: @rodrigopaulino why did you remove this? DDMTemplateModelDocumentContributor is still indexing this attribute. Any reason to do this?

Comment (rodrigopaulino) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/search/spi/model/index/contributor/DDMStructureModelDocumentContributor.java]: I removed it because it doesn't make sense since DDMStructure doesn't have a resourceClassNameId attribute. This was my first solution, but I saw that after. That's why I sent you this new solution.
